src/main/groovy/edu/oregonstate/mist/metaxe/XEAppsResource.groovy

```diff
@@ -0,0 +1,139 @@
+package edu.oregonstate.mist.metaxe
+
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+
+import com.codahale.metrics.annotation.Timed
+import javax.annotation.security.PermitAll
+import javax.ws.rs.GET
+import javax.ws.rs.NotFoundException
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.UriBuilder
+import java.util.regex.Pattern
+
+@Path("/xeapps")
+@Produces(MediaType.APPLICATION_JSON)
+@groovy.transform.TypeChecked
+class XEAppsResource extends Resource {
+    private XEAppDAO dao
+    private URI myEndpointUri
+
+    private final String JSONAPI_TYPE = "xeapp"
+
+    XEAppsResource(XEAppDAO dao, URI endpointUri) {
+        this.dao = dao
+        this.myEndpointUri = endpointUri
+        this.endpointUri = endpointUri
+    }
+
+    // Get a specific application by ID.
+    @Timed
+    @GET
+    @Path("{id}")
+    @PermitAll
+    ResultObject getById(@PathParam("id") String id) {
+        id = sanitize(id)
+
+        ESResult es = this.dao.getById(id)
+        if (es == null) {
+            throw new NotFoundException()
```

Comment:
Per the recommendation in the [dropwizard docs](http://www.dropwizard.io/1.0.6/docs/manual/core.html#responses), getById returns a domain object (ResultObject) rather than a Response. Note that NotFoundException will be caught by NotFoundExceptionMapper, which will return Resource.notFound().